File: Python/counting-valleys.py
# https://www.hackerrank.com/challenges/counting-valleys/problem

def countingValleys(steps, path):
    print(steps)
    print(path)
    elevation = 0
    valleyCount = 0
    for step in path:
        if step == "U":
            elevation += 1
        else:
            elevation -= 1
        print(elevation)
        
        if elevation == 0 and step == "U":
            valleyCount += 1
        else:
            pass
        
    return valleyCount

File: Javascript/counting-valleys.js
// https://www.hackerrank.com/challenges/counting-valleys/problem

function countingValleys(steps, path) {
    console.log(steps)
    console.log(path)
    var elevation = 0
    var valleyCount = 0
    for (var step of path) {
        console.log(step)
        if (step == "U"){
            elevation += 1
            console.log("Elevation up")
        }
        else {
            elevation -= 1
            console.log("Elevation down")
        }
        
        if (elevation == 0 && step == "U") {
            valleyCount += 1
        }
    }
    
    return valleyCount
}